src/globalstate.h

```diff
@@ -18,9 +18,30 @@
 #include <src/gasman.h>
 
 #define MAXPRINTDEPTH 1024L
+#define TLS_NUM_EXTRA 256
 
-typedef struct GlobalState
+typedef struct GAPState
 {
+#if defined(HPCGAP)
+  int threadID;
+  void *threadLock;
+  void *threadSignal;
+  void *acquiredMonitor;
+  unsigned multiplexRandomSeed;
+  void *currentRegion;
+  void *threadRegion;
+  void *traversalState;
+  Obj threadObject;
+  Obj tlRecords;
+  Obj lockStack;
+  int lockStackPointer;
+  Obj copiedObjs;
+  Obj interruptHandlers;
+  void *CurrentHashLock;
+  char *CurrFuncName;
+  int DisableGuards;
+#endif
```

Comment:
I think we should just merge the structs, make sure everything works, then separate things out again. I have too often tried to change too many things at the same time, and gotten myself into a mess.